### ISSUE TITLE
NODE_ENV is not being set to `production` during application launch

### ISSUE THREAD

**Issue by johngrogg** [cloudfoundry/nodejs-buildpack#90]
When deploying an application with v1.5.34, the NODE_ENV variable is set to `development`. You can reproduce this by simply adding `console.log(process.env.NODE_ENV);` to a script that runs during startup of the application and then reviewing the logs.

This causes significant problems for any node scripts that exhibit different behavior based on NODE_ENV being equal to `production`.

What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
{
   "name": "",
   "build": "",
   "support": "https://support.run.pivotal.io",
   "version": 0,
   "description": "Cloud Foundry sponsored by Pivotal",
   "authorization_endpoint": "https://login.run.pivotal.io",
   "token_endpoint": "https://uaa.run.pivotal.io",
   "min_cli_version": "6.22.0",
   "min_recommended_cli_version": "latest",
   "api_version": "2.82.0",
   "app_ssh_endpoint": "ssh.run.pivotal.io:2222",
   "app_ssh_host_key_fingerprint": "e7:13:4e:32:ee:39:62:df:54:41:d7:f7:8b:b2:a7:6b",
   "app_ssh_oauth_client": "ssh-proxy",
   "routing_endpoint": "https://api.run.pivotal.io/routing",
   "doppler_logging_endpoint": "wss://doppler.run.pivotal.io:443",
   "user": "f91c356d-779f-4bbe-8dcd-379054b57ea8"
}
cf version 6.26.0+9c9a261fd.2017-04-06

What version of the buildpack you are using?
v1.5.34

If you were attempting to accomplish a task, what was it you were attempting to do?
Push a node app that relies on the `process.env.NODE_ENV` being equal to `production` to shape database credentials appropriately for the production environment.

What did you expect to happen?
`process.env.NODE_ENV` to equal `production`

What was the actual behavior?
`process.env.NODE_ENV` equaled `development`

Please confirm where necessary:
* [ ] I have included a log output
* [ ] My log includes an error message
* [ ] I have included steps for reproduction

**Comment by sclevine** [comment]
Are you sure you don't just have `NODE_ENV` set to `development` on your app?
See: https://github.com/cloudfoundry/nodejs-buildpack/blob/v1.5.34/lib/environment.sh#L5

**Comment by johngrogg** [comment]
Yes. The User Provided Environment Variables are empty and all environment variables are being set as part of the CF runtime and/or buildpack.

Using v1.5.32 of the buildpack `NODE_ENV` is set to `production` with the only difference in the push command being the buildpack version.

**Comment by sclevine** [comment]
Hi @johngrogg,

While I cannot replicate `NODE_ENV` being set to `development` in either version, I can replicate it not being set at all in v1.5.34. We should be able to cut a new release with a fix on Tuesday.

In v1.5.32:
```
vcap@fe6f1dd5-9cc9-4131-57fc-7f8b:~$ ps aux | grep node
vcap          18  1.3  0.0 878604 30156 ?        S<sl 20:49   0:00 node server.js
vcap@fe6f1dd5-9cc9-4131-57fc-7f8b:~$ xargs --null --max-args=1 echo < /proc/18/environ | grep NODE_ENV
NODE_ENV=production
vcap@fe6f1dd5-9cc9-4131-57fc-7f8b:~$
```

In v1.5.34:
```
vcap@bdef80fe-96fd-49db-4836-49a9:~$ ps aux | grep node
vcap          14  0.5  0.0 879052 30460 ?        S<sl 20:42   0:00 node server.js
vcap@bdef80fe-96fd-49db-4836-49a9:~$ xargs --null --max-args=1 echo < /proc/14/environ |grep NODE_ENV
vcap@bdef80fe-96fd-49db-4836-49a9:~$
```

Thanks for the report!

**Comment by sclevine** [comment]
Fixed in [v1.5.35](https://github.com/cloudfoundry/nodejs-buildpack/releases/tag/v1.5.35).
